launcher/src/main/java/org/apache/spark/launcher/CommandBuilderUtils.java

```diff
@@ -334,6 +334,18 @@ static void addPermGenSizeOpt(List<String> cmd) {
   }
 
   /**
+   * Gets the OutOfMemoryError option for Spark if the user hasn't set it.
+   */
+  public static String getOutOfMemoryErrorArgument(List<String> cmd) {
```

Comment:
The pattern ExecutorRunnable uses is that the return value is added explicitly to Java opts.

I think being consistent with the pattern is fine.